src/main/java/org/opensearch/knn/index/remote/RemoteIndexHTTPClient.java

```diff
@@ -124,13 +124,45 @@ private HttpPost getHttpPost(String jsonRequest) {
     }
 
     /**
-    * Await the completion of the index build by polling periodically and handling the returned statuses until timeout.
-    * @param remoteBuildResponse containing job_id from the server response used to track the job
-    * @return RemoteStatusResponse containing the path to the completed index
-    */
+     * Await the completion of the index build using a {@link RemoteIndexPoller}.
+     * @param remoteBuildResponse containing job_id from the server response used to track the job
+     * @return RemoteBuildStatusResponse containing the path to the completed index
+     */
     @Override
-    public RemoteStatusResponse awaitVectorBuild(RemoteBuildResponse remoteBuildResponse) {
-        throw new NotImplementedException();
+    public RemoteBuildStatusResponse awaitVectorBuild(RemoteBuildResponse remoteBuildResponse) throws InterruptedException, IOException {
```

Comment:
awaitVectorBuild should accept a input object RemoteBuildStatusRequest rather than `RemoteBuildResponse`